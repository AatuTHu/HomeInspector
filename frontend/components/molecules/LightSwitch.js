import { View, Text } from 'react-native'
import { styles } from "../../styles/styles"
import { useState } from 'react'
import Buttons from '../atoms/Buttons'
import variables from '../../env'

export default function LightSwitch() {

  const { apiKey,lightURL } = variables
  const [lights, setLights] = useState(false)

    const lightSwitch = async() => {
        
        let mode = lights ?  1 : 0
    
        const response = await fetch(lightURL, {
          method: "POST",
          headers: { "Content-Type": "application/x-www-form-urlencoded" },
          body: `apiKey=${apiKey}&lightMode=${mode}`,
        })
    
        if(response.status === 202) {
          setLights(!lights)
        }
      }

  return (
    <View style = { styles.buttonRowContainer }>
        <Buttons text = "lights" type = "start" onPress={ lightSwitch }/>
      <View style = { styles.statusBox }>
          <Text style = { styles.subHeadingText }>{lights ? (<Text style = {{"color" : "yellow"}}>On</Text>) : (<Text>Off</Text>)}</Text>
      </View>
  </View>
  )
}