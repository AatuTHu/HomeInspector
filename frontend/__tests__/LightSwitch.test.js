import { render, fireEvent, waitFor } from '@testing-library/react-native';
import LightSwitch from '../components/molecules/LightSwitch';

// Mock the fetch function globally
global.fetch = jest.fn(() =>
  Promise.resolve({
    status: 200,
    json: () => Promise.resolve({}),
  })
);

describe('LightSwitch', () => {
afterEach(() => {
  jest.clearAllMocks();
});

it('should render correctly', () => {
  const { getByText } = render(<LightSwitch lights={false} setLights={jest.fn()} />);
  expect(getByText('Off')).toBeTruthy();
});

it('should turn lights ON and button is pressed', async () => {

  global.fetch = jest.fn(() =>
      Promise.resolve({
        status: 200,
        json: () => Promise.resolve({}),
  }));

  setLights = jest.fn()
  const { getByText } = render(<LightSwitch lights={true} setLights={setLights} />);
  fireEvent.press(getByText('lights'));
  await waitFor(() => 
    expect(getByText('On')).toBeTruthy(),
  )}
)


it('should turn lights OFF and button is pressed', async () => {
  global.fetch = jest.fn(() =>
      Promise.resolve({
        status: 304,
        json: () => Promise.resolve({}),
  }));

  setLights = jest.fn()
  const { getByText } = render(<LightSwitch lights={false} setLights={setLights} />);
  fireEvent.press(getByText('lights'));
  await waitFor(() => 
    expect(getByText('Off')).toBeTruthy()
  )
})

});
