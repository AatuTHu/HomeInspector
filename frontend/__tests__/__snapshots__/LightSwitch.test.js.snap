// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lightSwitch renders correctly 1`] = `
<View
  style={
    {
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 20,
      "marginTop": 1,
      "width": "95%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "lightgrey",
        "borderRadius": 20,
        "borderWidth": 1,
        "opacity": 1,
        "padding": 5,
        "width": "50%",
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      lights
    </Text>
  </View>
  <View
    style={
      {
        "backgroundColor": "lightgrey",
        "borderRadius": 22,
        "justifyContent": "center",
        "width": "30%",
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      <Text>
        Off
      </Text>
    </Text>
  </View>
</View>
`;
